### DISCUSSION TITLE
Sketch for multiple camera live stitching setup

### DISCUSSION THREAD

**Discussion by shuckerino** [OpenStitching/stitching#116] [Q&A]
Hello, as I have already shared in another discussion, I need to realise live video stitching with multiple (4) cameras. The plan is to mount 4 cameras on top of a model car within a 90 degree angle. Each camera has a field of view of around 110 degrees and they are positioned beside each other on top of the roof. I now want to develop a python script that can stitch the images of the 4 cameras together with a decent frame rate (the optimal goal would be 30 fps). 
At the end I want to have a 360 degree panorama image (video) that I can live stream to UnrealEngine (there It will be displayed and used vor VR)

The following is a rough sketch (hope it is enough to display our idea):
![image](https://github.com/OpenStitching/stitching/assets/73401679/52e67400-3c97-4a4e-96cf-c2679b5e7255)

My most important question is if this repository could technically handle this kind of a load (because its Python) or if this would not work with a decent frame rate (just want to hear some opinions or experiences from you). Otherwise I need to look for alternatives (C++ etc). 

Thanks for your help!

**Comment by lukasalexanderweber** [comment]
Do you already have the cameras? Could you share test images of the setup? Then you could get some Timing Benchmarks. OpenCV is C++ and this is only a wrapper (running C in the Background), so it wont get that much faster. 

Also check out https://github.com/OpenStitching/stitching/discussions/89, there are some problems integrating results in VR/360° Environments

**Reply by shuckerino** [reply]
Unfortunately, I have just been assigned to this task and dont have the real setup available. My current task is to just try to make live image stitching work for 2 webcams (but of course I want to make it already scalable to the 4 cameras). I will try to get test images from the real setup ASAP. 

Good to know that C++ would not be much faster because of OpenCV basically being C(++).

And also thanks for mentioning the related issue, I will take a look at it and maybe come back with some more information later.

**Comment by lukasalexanderweber** [comment]
@shuckerino with ad961e80da59ceb1ca4cf8dbf5747d606092e579 it's now possible to stitch loaded images directly
